velox/vector/BaseVector.cpp

```diff
@@ -575,6 +575,16 @@ VectorPtr newConstant<TypeKind::LONG_DECIMAL>(
     velox::memory::MemoryPool* pool) {
   // LongDecimal variant is not supported to create
   // constant vector.
+  if (value.isNull()) {
+    return std::make_shared<ConstantVector<LongDecimal>>(
+        pool,
+        size,
+        value.isNull(),
+        LONG_DECIMAL(LongDecimalType::kMaxPrecision, 0),
```

Comment:
We need to use the type specified by the caller of BaseVector::createNullConstant.